framework/wazuh/core/config/client.py

```diff
@@ -0,0 +1,175 @@
+        ------
+        FileNotFoundError
+            If the configuration file does not exist.
+        """
+        if cls._config is None:
+            if not os.path.exists(WAZUH_SERVER_YML):
+                raise FileNotFoundError(f"Configuration file not found: {WAZUH_SERVER_YML}")
+            with open(WAZUH_SERVER_YML, 'r') as file:
+                config_data = yaml.safe_load(file)
+                cls._config = Config(**config_data)
+
+    @classmethod
+    def get_comms_api_config(cls) -> CommsAPIConfig:
+        """Retrieve the communications API configuration.
+
+        Loads the configuration if it has not been loaded yet.
+
+        Returns
+        -------
+        CommsAPIConfig
+            The communications API configuration.
+        """
+        if cls._config is None:
+            cls.load()
+
+        return cls._config.communications_api
+
+    @classmethod
+    def get_management_api_config(cls) -> ManagementAPIConfig:
+        """Retrieve the management API configuration.
+
+         Loads the configuration if it has not been loaded yet.
+
+        Returns
+        -------
+        ManagementAPIConfig
+            The management API configuration.
+        """
+        if cls._config is None:
+            cls.load()
+
+        return cls._config.management_api
+
+    @classmethod
+    def get_indexer_config(cls) -> IndexerConfig:
+        """Retrieve the indexer configuration.
+
+        Loads the configuration if it has not been loaded yet.
+
+        Returns
+        -------
+        IndexerConfig
+            The indexer configuration.
+        """
+        if cls._config is None:
+            cls.load()
+
+        return cls._config.indexer
+
+    @classmethod
+    def get_engine_config(cls) -> EngineConfig:
+        """Retrieve the engine configuration.
+
+        Loads the configuration if it has not been loaded yet.
+
+        Returns
+        -------
+        EngineConfig
+            The engine configuration.
+        """
+        if cls._config is None:
+            cls.load()
+
+        return cls._config.engine
+
+    @classmethod
+    def get_server_config(cls) -> ServerConfig:
+        """Retrieve the server configuration.
+
+        Loads the configuration if it has not been loaded yet.
+
+        Returns
+        -------
+        ServerConfig
+            The server configuration.
+        """
+        if cls._config is None:
+            cls.load()
+
+        return cls._config.server
+
+    @classmethod
+    def get_internal_server_config(cls) -> ServerSyncConfig:
+        """Retrieve the internal server configuration.
+
+        Loads the configuration if it has not been loaded yet.
+
+        Returns
+        -------
+        ServerSyncConfig
+            The internal server configuration.
+        """
+        if cls._config is None:
+            cls.load()
+
+        return cls._config.server.get_internal_config()
+
+    @classmethod
+    def update_security_conf(cls, config: dict):
```

Comment:
Add unit test for this new function

Reply:
Added in b90f3a2e3d0f894e820916c18fb6621a6d5fb34b.